ListaDeTarefas/ViewModel/AdicionaListaViewController.swift

```diff
@@ -0,0 +1,127 @@
+//
+//  AdicionaListaViewController.swift
+//  ListaDeTarefas
+//
+//  Created by Angélica Andrade de Meira on 25/04/22.
+//
+
+import UIKit
+import CoreData
+
+class AdicionaListaViewController: UIViewController, UITextFieldDelegate, TelaInicialTableViewControllerDelegate, UINavigationControllerDelegate {
+    func chamaRecuperaListas() {
+//        let telaInicial: TelaInicialTableViewController
+//        telaInicial.recuperaListas()
+    }
+    
+    var contexto: NSManagedObjectContext!
+    var listaSelecionada: NSManagedObject?
+    
+    // MARK: - View code
+    
+    private lazy var botaoOk: UIBarButtonItem = {
+        let view = UIBarButtonItem()
+        view.title = "Ok"
+        view.action = #selector(ok)
+        return view
+      }()
+    
+    private lazy var botaoCancelar: UIBarButtonItem = {
+        let view = UIBarButtonItem()
+        view.title = "Cancelar"
+        view.action = #selector(cancelar)
+        return view
+      }()
+    
+    private lazy var campoDescricao: UITextField = {
+        let view = UITextField()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.borderStyle = .roundedRect
+        view.placeholder = "Insira o nome da lista"
+        view.returnKeyType = .done
+        view.becomeFirstResponder()
+        return view
+      }()
+    
+    override func loadView() {
+        super.loadView()
+        
+        self.view.backgroundColor = .white
+        self.navigationItem.rightBarButtonItems = [botaoOk]
+        self.navigationItem.leftBarButtonItems = [botaoCancelar]
+        self.view.addSubview(campoDescricao)
+        
+        if listaSelecionada != nil {
+            self.title = "Nova lista"
+        }else{
+            self.title = "Editar lista"
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setup()
+
+        campoDescricao.topAnchor.constraint(equalTo: self.view.topAnchor, constant: 60).isActive = true
+        campoDescricao.trailingAnchor.constraint(equalTo: self.view.trailingAnchor, constant: -20).isActive = true
+        campoDescricao.leadingAnchor.constraint(equalTo: self.view.leadingAnchor, constant: 20).isActive = true
+        campoDescricao.delegate = self
+        
+        let appDelegate = UIApplication.shared.delegate as! AppDelegate
+        contexto = appDelegate.persistentContainer.viewContext
+    }
+    
+    @objc func ok() {
+        
+        if listaSelecionada == nil {
+            salvarNovaLista()
+        } else {
+            atualizarNomeDaLista()
+        }
+        self.navigationController?.dismiss(animated: true) { [unowned self] in
```

Comment:
Pode explicar por que usou unowned aqui? Quando se usa unowned e o elemento retido é desalocado, ele causa um crash no aplicativo

Reply:
Você disse que em toda closure precisa usar o unowned self, se for usar o self dentro dela. Já corrigi.

Reply:
Eu falei `weak self`

Reply:
ok